src/match.js

```diff
@@ -1,19 +1,24 @@
+var _curry2 = require('./internal/_curry2');
+var compose = require('./internal/compose');
+var defaultTo = require('./defaultTo');
 var invoker = require('./invoker');
 
 
 /**
- * Tests a regular expression against a String
+ * Tests a regular expression against a String. Note that this function
+ * will return an empty array when there are no matches. This differs
+ * from [`String.prototype.match`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match)
+ * which returns `null` when there are no matches.
  *
  * @func
  * @memberOf R
  * @category String
- * @sig RegExp -> String -> [String] | null
+ * @sig RegExp -> String -> [String | Undefined]
  * @param {RegExp} rx A regular expression.
  * @param {String} str The string to match against
- * @return {Array} The list of matches, or null if no matches found.
- * @see R.invoker
+ * @return {Array} The list of matches.
  * @example
  *
  *      R.match(/([a-z]a)/g, 'bananas'); //=> ['ba', 'na', 'na']
  */
-module.exports = invoker(1, 'match');
+module.exports = _curry2(compose(defaultTo([]), invoker(1, 'match')));
```

Comment:
Am I right, that if someone evil will change resulting array then everybody will suffer because of shared default array pointer?

Example:
```js
R.match(/a/, '')[1] = 'fail';
R.match(/a/, ''); // [, 'fail']
```
Reproducing on Ramda site.

Sure, there is purity around all Ramda, but it is not guaranteed in the wild and someone could get a hard pain in the ass debugging.